src/rgw/rgw_data_sync.cc

```diff
@@ -1294,17 +1296,24 @@ class RGWDataSyncShardCR : public RGWCoroutine {
           return set_cr_error(retcode);
         }
         datalog_marker = shard_info.marker;
+        remote_trimmed = 0;
 #define INCREMENTAL_MAX_ENTRIES 100
 	ldout(sync_env->cct, 20) << __func__ << ":" << __LINE__ << ": shard_id=" << shard_id << " datalog_marker=" << datalog_marker << " sync_marker.marker=" << sync_marker.marker << dendl;
 	if (datalog_marker > sync_marker.marker) {
           spawned_keys.clear();
+          if (sync_marker.marker.empty())
+            remote_trimmed = 2; //remote data log shard might be trimmed;
```

Comment:
are the 0..1 values of remote_trimmed actually ordinals?  if not, I think they should be constants or enumeration values